python/tank/deploy/tank_commands/update.py

```diff
@@ -149,80 +217,92 @@ def run_interactive(self, log, args):
+    :param suppress_prompts: If True, run without prompting
+    :param preserve_yaml: Indicates that a comment preserving yaml parser 
+                          should be used.
+    :param external: Path to external config to operate on
     """
+    
     pc = tk.pipeline_configuration
-
-    if env_name is None:
-        env_names_to_process = pc.get_environments()
-    else:
-        env_names_to_process = [env_name]
-
-    # check engines and apps
-    items = []
-    for env_name in env_names_to_process:
-        
-        # (AD) - Previously all environments were loaded before processing but 
-        # this could lead to errors if an update doesn't know about previous 
-        # updates to the same share files (includes)
-        env = pc.get_environment(env_name)
+    
+    updated_items = []
+    
+    if external:
         
+        # try to load external file
+        if not os.path.exists(external):
+            log.error("Cannot find external config %s" % external)
+            return
+            
+        env_path = os.path.join(external, "env")
+            
+        if not os.path.exists(env_path):
+            log.error("Cannot find environment folder '%s'" % env_path)
+            return
+            
+        # find all environment files
+        log.info("Looking for matching environments in %s:" % env_path)
         log.info("")
-        log.info("")
-        log.info("======================================================================")
-        log.info("Environment %s..." % env.name)
-        log.info("======================================================================")
-        log.info("")
+        env_filenames = []
+        for filename in os.listdir(env_path):
+            if filename.endswith(".yml"):
+                if env_name is None or ("%s.yml" % env_name) == filename: 
+                    # matching the env filter (or no filter set)
+                    log.info("> found %s" % filename) 
+                    env_filenames.append(os.path.join(env_path, filename))
         
-        if engine_instance_name is None:
-            # process all engines
-            engines_to_process = env.get_engines()
-        else:
-            # there is a filter! Ensure the filter matches something
-            # in this environment
-            if engine_instance_name in env.get_engines():
-                # the filter matches something in this environment
-                engines_to_process = [engine_instance_name] 
-            else:
-                # the item we are filtering on does not exist in this env
-                engines_to_process = []
-        
-        for engine in engines_to_process:
-            items.append( _process_item(log, suppress_prompts, tk, env, engine) )
-            log.info("")
+        # now process them one after the other
+        for env_filename in env_filenames: 
+            env_obj = WritableEnvironment(env_filename, pc)
```

Comment:
we discussed on hipchat and decided to go with the existing flag in lieu of anything better - at the end of the day, this is a temp transitionary thing anyways.